tests/e2e/vmgroupmisc_test.go

```diff
@@ -0,0 +1,261 @@
+	isVMPartofVg := admincli.IsVMInVmgroup(s.config.EsxHost, s.config.DockerHostNames[0], vmGroupName)
+	c.Assert(isVMPartofVg, Equals, false, Commentf("Unexpected Behavior: VM %s belong to vmgroup %s .", s.config.DockerHostNames[0], vmGroupName))
+
+	admincli.DeleteVMgroup(s.config.EsxHost, vmGroupName)
+	// Verify vmgroup does not exist
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, false, Commentf("Failed to delete the vmgroup [%s] .", vmGroupName))
+
+	// Removing the DB at the end of suite
+	admincli.ConfigRemove(s.config.EsxHost)
+
+	// Verifying DB successfully removed
+	c.Assert(admincli.GetDBmode(s.config.EsxHost), Equals, admincliconst.DBNotConfigured, Commentf("Failed to remove the DB mode on ESX -  .", s.config.EsxHost))
+
+}
+
+var _ = Suite(&vgBasicSuite{})
+
+// Verify vmgroup for orphaned volumes is specified as N/A
+// 1. Create a vmgroup and add vm to it.
+// 2. Verify vmgroup was successfully created.
+// 3. Create a volume from vm-2.
+// 4. Verify volumes vmgroup from esx.
+// 5. Remove vm-2 from vmgroup.
+// 6. Verify volume is not visible from vm-2.
+// 7. Delete the tenant.
+// 8. Again verify volumes vmgroup is listed as N/A.
+func (s *vgBasicSuite) TestVGNameForOrphanedVolumes(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume from vmgroup's vm
+	out, err := dockercli.CreateVolume(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Check if volume was successfully created
+	isAvailable := verification.CheckVolumeAvailability(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(isAvailable, Equals, true, Commentf("Volume %s is not available after creation", s.volumeNames[0]))
+
+	// Verify if volume exists in vmgroup
+	isVolInVmgroup := admincli.IsVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[0], vmGroupName)
+	c.Assert(isVolInVmgroup, Equals, true, Commentf("Volume [%s] does not belong to vmgroup [%s]", s.volumeNames[0], vmGroupName))
+
+	admincli.RemoveVMFromVMgroup(s.config.EsxHost, vmGroupName, s.config.DockerHostNames[0])
+
+	// Check if volume was not visible from vm since it does not belong to vmgroup
+	isAvailable = verification.CheckVolumeAvailability(s.config.DockerHosts[0], s.volumeNames[0])
+	c.Assert(isAvailable, Equals, false, Commentf("Unexpected Behavior: Volume %s belonging to "+
+		"vmgroup [%s] is visible from host [%s] which does not belong to the same vmgroup", s.volumeNames[0], vmGroupName, s.config.DockerHosts[0]))
+
+	// Now delete the vmgroup
+	admincli.DeleteVMgroup(s.config.EsxHost, vmGroupName)
+
+	// Verify vmgroup does not exist
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, false, Commentf("Failed to delete the vmgroup [%s] .", vmGroupName))
+
+	// Verify vmgroup for volume is N/A
+	isVolInVmgroup = admincli.IsVolumeExistInVmgroup(s.config.EsxHost, s.volumeNames[0], admincliconst.NotApplicable)
```

Comment:
Fully agree, that keeps happening. But as code author I'd suggest you going over review comments and figuring out exactly whats really needed vs. good to have or specific code style. And only then making changes based on your judgement.

In this case a function like IsVolumeExistingInVmgroup() should get a _vmgroup_ name always and hence a special case like _NotApplicable_ doesn't seem correct. When we define an API, the params have a meaning and the values passed should reflect that. So defining a constant for a _NullVmgroup_ as N/A and using that keeps the code understandable. And if no one else needs a NullVmgroup in their tests lets keep it locally. Thats the reasoning behind doing it this way.